File: functions/src/ops/model/appinfo.ts
/**
 * Copyright © 2020 WorldRIZe. All rights reserved.
 */
export interface WRAppInfo {
  /**
   * 現在のバージョン
   */
  currentVersion: string

  /**
   * 必要なバージョン
   */
  requireVersion: string

  /**
   * iOS メンテナンス中か
   */
  isIOsAppAvailable: boolean

  /**
   * Android メンテナンス中か
   */
  isAndroidAppAvailable: boolean
}

File: lib/domain/system/model/app_info.dart
// Copyright © 2020 WorldRIZe. All rights reserved.

import 'dart:io';

import 'package:flutter/foundation.dart';
import 'package:json_annotation/json_annotation.dart';

part 'app_info.g.dart';

@JsonSerializable(explicitToJson: true, anyMap: true)
class AppInfo {
  AppInfo({
    required this.currentVersion,
    required this.requireVersion,
    required this.isIOsAppAvailable,
    required this.isAndroidAppAvailable,
  });

  String currentVersion;

  String requireVersion;

  bool isIOsAppAvailable;

  bool isAndroidAppAvailable;

  factory AppInfo.fromJson(Map<String, dynamic> json) {
    return _$AppInfoFromJson(json);
  }

  Map<String, dynamic> toJson() => _$AppInfoToJson(this);

  // TODO: versionのチェックをする
  bool get isValid {
    if (Platform.isIOS && isIOsAppAvailable) {
      return true;
    } else if (Platform.isAndroid && isAndroidAppAvailable) {
      return true;
    }
    return false;
  }
}
